lib/encoder.js

```diff
@@ -90,116 +145,88 @@ Encoder.prototype.encode = function (value, typeInfo) {
   if (value === null || value === undefined) {
     return null;
   }
+  /** @type {{code: Number, info: object}} */
+  var type = {
+    code: null,
+    info: null
+  };
   if (typeInfo) {
     if (typeof typeInfo === 'number') {
-      //noinspection JSValidateTypes
-      typeInfo = {type: typeInfo};
+      type.code = typeInfo;
     }
     else if (typeof typeInfo === 'string') {
-      typeInfo = dataTypes.getByName(typeInfo);
+      type = dataTypes.getByName(typeInfo);
     }
-    if (typeof typeInfo.type !== 'number') {
+    if (typeof typeInfo.code === 'number') {
```

Comment:
These two if statements probably be in the else if statement above at line 157 since we know typeInfo.code !== number and type.code === number  since typeInfo is a number